### ISSUE TITLE
Avocado doen't recognize tests depending on package layout

### ISSUE THREAD

**Issue by beraldoleal** [avocado-framework/avocado#4611]
Let's say that I have a package with the following structure:

```
.
├── foo/
│   ├── bar.py
│   └── __init__.py
└── tests/
    └── test_one.py
```

With the following contents:

```python
# foo/bar.py
from avocado import Test

class BarTest(Test):
    def setUp(self):
        """My custom setup here"""
        self.bar = 'foo'
```

```python
# tests/test_one.py
# from foo.bar import BarTest 

class TestOne(BarTest):
    def test_one(self):
        self.assertTrue(1)
```

After setting a proper PYTHONPATH, the code above gives me this error:

```
$ avocado run --test-runner='nrunner' tests/test_one.py 
Could not resolve references: tests/test_one.py
```
Now, if I move the `foo/bar.py` code to the package `__init__.py` file, it will work fine (after changing the import on `test_one.py`.

IMO, we should support both.

**Comment by clebergnu** [comment]
Closing as a duplicate of #4625

**Comment by beraldoleal** [comment]
@clebergnu when duplicate, we could close the newest right?

**Comment by clebergnu** [comment]
This is a actually a similar, but not identical, issue.  Reopening.

**Comment by clebergnu** [comment]
Fixed by #4718
